### ISSUE TITLE
difference in result

### ISSUE THREAD

**Issue by Carrotor116** [Algy/fast-slic#3]
Your implement is real fast. While I notice that there are some difference between  the result of  your implement and ones of `skimage.segmentation.slic`. I wonder whether it's working correctly.
Here is my code and result:
* slic of `fast-slic`
```
import numpy as np
from PIL import Image
from fast_slic import Slic
from skimage.segmentation import mark_boundaries

with Image.open("fish.jpg") as f:
    image = np.array(f)

slic = Slic(num_components=200, compactness=6)
assignment = slic.iterate(image)  # Cluster Map
out = mark_boundaries(image, assignment) * 255
out = Image.fromarray(out.astype('uint8'))
out.save('fish_sp.jpg')
```
![fish_sp](https://user-images.githubusercontent.com/22851737/56426025-440b3d00-62e9-11e9-8db4-e771824e4457.jpg)

* slic of `skimage`
```
from PIL import Image
from skimage import io
from skimage.segmentation import slic, mark_boundaries

img = io.imread("fish.jpg")

segments = slic(img, n_segments=200, compactness=6)
out = mark_boundaries(img, segments) * 255
out = Image.fromarray(out.astype('uint8'))
out.save('fish_sp_skimage.jpg')
```
![fish_sp_skimage](https://user-images.githubusercontent.com/22851737/56426043-54bbb300-62e9-11e9-9119-8af6d34c6930.jpg)

And with config `num_components=200`, `compactness=6`, the result still different from one in [README.md](https://github.com/Algy/fast-slic/blob/master/README.md) 

![result in README.md](https://user-images.githubusercontent.com/2352985/55978839-c5504780-5ccb-11e9-9820-d8ddf950f230.png).

here is my input `fish.jpg` 
![fish](https://user-images.githubusercontent.com/22851737/56426204-fcd17c00-62e9-11e9-93cf-60ed17a3014e.jpg)

so how can I get the same result in your demo :)

**Comment by Algy** [comment]
Thanks for your feedback. Try change it to 64 and you will see the same result. 
It is due to change of meaning of `compactness` parameter. Originally, for performance, compactness was actually named `compactness_shift`, which is power of 2 of compactness (You know, shift instruction is faster than multiplication :). But after having a lot of change of code base I didn't need this micro-optimization anymore, so I renamed it to `compactness` but think I forgot to change parameter value in README. I'm going to work hard in this weekend so I can keep up README.md with the recent modification. Also, I'm adding tests for consistent development.

**Comment by Algy** [comment]
The result is also quite different from that of the original SLIC implementation. This is intended.  Actually, it uses different distance measure, heavy caching, floating-point quantization, and some mathematically ugly but super-fast computation. As it is intended for use in video streaming, a little clumsiness didn't matter at all. It would be very helpful if you could share an idea to improve the accuracy :) (without hurting performance too much). thanks.

**Comment by Algy** [comment]
All issues mentioned above are all fixed in release 0.3.1. plz upgrade fast-slic to 0.3.1. I'm closing the issue.
